lightningd/peer_control.c

```diff
@@ -2220,7 +2220,7 @@ void setup_peers(struct lightningd *ld)
 struct htlc_in_map *load_channels_from_wallet(struct lightningd *ld)
 {
 	struct peer *peer;
-	struct htlc_in_map *unconnected_htlcs_in = tal(ld, struct htlc_in_map);
+	struct htlc_in_map *unconnected_htlcs_in;
```

Comment:
Moving this allocation further down does not have any change in the memleak logic, since we can't be interrupted inbetween this declaration and the below initialization (all code is sync so no way a `dev-memleak` could catch us in the middle).

Reply:
yep I was in doubt about the existing workflow regarding `fatal`, I guess this PR is useless